src/main/java/org/spongepowered/api/item/inventory/ItemStack.java

```diff
@@ -131,6 +135,8 @@ static ItemStack of(ItemType itemType, int quantity) {
          */
         Builder itemType(ItemType itemType);
 
+        ItemType getCurrentItem();
```

Comment:
@Zidane The feature was requested by @Wolfzein for adding some precondition based data application.